url/Cargo.toml

```diff
@@ -22,17 +22,25 @@ serde_json = "1.0"
 bencher = "0.1"
 
 [dependencies]
-form_urlencoded = { version = "1.2.1", path = "../form_urlencoded" }
-idna = { version = "0.5.0", path = "../idna" }
-percent-encoding = { version = "2.3.1", path = "../percent_encoding" }
-serde = { version = "1.0", optional = true, features = ["derive"] }
+form_urlencoded = { version = "1.2.1", path = "../form_urlencoded", default-features = false, features = ["alloc"] }
+idna = { version = "0.5.0", path = "../idna", default-features = false, features = ["alloc"] }
+percent-encoding = { version = "2.3.1", path = "../percent_encoding", default-features = false, features = ["alloc"] }
+serde = {version = "1.0", optional = true, features = ["derive"]}
+no-std-net = { version = "0.6.0", default-features = false, optional = true }
 
 [features]
-default = []
+default = ["std"]
+std = ["idna/std", "percent-encoding/std", "form_urlencoded/std", "alloc"]
+alloc = []
 # Enable to use the #[debugger_visualizer] attribute. This feature requires Rust >= 1.71.
 debugger_visualizer = []
 # Expose internal offsets of the URL.
 expose_internals = []
+# For no_std: Allow the use of no_std_net instead of nightly
+no_std_net = ["no-std-net"]
+# UNSTABLE FEATURES (requires Rust nightly)
+# For no_std: Use errors_in_core and net_in_core
+unstable = ["idna/unstable"]
```

Comment:
https://github.com/rust-lang/rust/issues/108443 is FCP-ing now, so it'll probably land in a few months - perhaps we should wait with this feature until then?

Reply:
Right now this PR also uses `error_in_core` (https://github.com/rust-lang/rust/issues/103765) but that part can be removed if we want to go for stable instead

Reply:
The value in that feature is that this could become a non-breaking change (other than bumping MSRV, which is probably acceptable).

Reply:
Update: `core::net` has landed in Rust 1.77.0